### ISSUE TITLE
Employing DeepCell in COSMX data

### ISSUE THREAD

**Issue by joaolsf** [ratschlab/DeepSpot#17]
Hi, I am starting to use DeepSpot/DeepCell in a COSMX/HE dataset. I plan to adapt the codes from the usage in Xenium to apply to COSMX. I've read around the GitHub issues in this repository and in the HE2ST repository but I still have few questions in the preprocessing step:
1- The count data should be raw counts or log-normalized? In the paper, it is mentioned that log-normalization in the preprocessing step but in one of the discussions here in the GitHub I read that raw counts should be used.
2- The cell diameter should be defined in a sample-basis, is that correct? The notebook suggests 20 but I assume this is based on the xenium lung sample. 
3- Should the cosmx data (cell coords in pxs) be pre-aligned to the HE image that will be used for feature extraction and training, correct?
4- How do you suggest to deal with cells in the border of the image? I had to edit the initial preprocessing codes to deal with them as some could not be cropped, but I wonder if you have a strategy you know works best.

Thank you!
Joao

**Comment by KalinNonchev** [comment]
Hi Jakob, thank you for your interest in our work!
1. We use log1p normalised data for training. 
2. Yes, it depends on your dataset and image resolution.
3. Yes, both modalities should be aligned so that you can train your model.
4. Dropping the cells at the borders based on the radius is the fast&dirty solution. Likely, you would lose a very small percentage of your overall training data. It is better to be consistent. 

Please let me know if you have further questions.

Best,

**Comment by joaolsf** [comment]
Hi @KalinNonchev, thanks for the clarification. I have been able to run the training and inference successfully. In this test, I just run prediction of sample I already have gene expression data, as I wanted to use it as my ground truth. However, I am not sure how to evaluate the performance of the prediction. Could you direct me to a resource in this repository or in the HEST repo to how I can benchmark the prediction vs the ground truth (the original expression)? Would be suitable a simple pearson correlation between predicted vs ground truth?

**Comment by KalinNonchev** [comment]
Hi @joaolsf, this depends a bit on your downstream task and what aspect of the prediction you care about. A common and reasonable starting point is to use correlation-based metrics, either:

- Per gene (compare expression across cells)
- Per cell (compare expression profiles across genes)

How would you like to use the predictions later?

Best,

**Comment by joaolsf** [comment]
HI @KalinNonchev, 

Yes, I have done a simple spearman correlation per gene and the correlation looks great - r around 0.98. 
The correlation per cell was a bit trickier because it had to rely on perfect alignment between the predicted and the original data - with that the initial correlation analysis I have done, generated quite low pearson correlation values and high RMSE. Hence, I asked you if there is a more proper way to evaluate this.

 I am comparing different tools for gene expression prediction from HE, and I quite like your tool. It is the fastest and easiest to implement so far. I have used iSCALE and they have a script to evaluate performance of the prediction which I tried to adapt here but not sure I am doing this correctly. 

Well, I would like to use the predictions for general spatial omics analysis just like you've showed in your paper, pathway analysis, etc, but also for cell typing. Again, for cell typing, iSCALE has an interesting method to penalize low "confidence" predicted markers and to use for marker-based cell annotation only genes passing this threshold, something I wonder would be interesting to have for downstream analysis with DeepSpot/DeepCell. Is this something you have been thinking to implement?

Please, let me know your suggestions about the evaluation of prediction performance and strategies for downstream analysis.

Thank you!

**Comment by KalinNonchev** [comment]
Hi @joaolsf, thanks for the detailed update, great to see the results so far!

These are interesting questions, but to give meaningful suggestions, I would need to better understand the specifics of your setup and analysis goals. Given that this goes a bit beyond the scope of the repository, it might make sense to discuss this in more depth offline. Feel free to reach out via email if you're interested.

**Comment by joaolsf** [comment]
Hi Kalin,

Happy to have a discussion offline. I am running DeepSpot at the moment comparing inference from the trained model in different samples. My plan is to do a cell annotation based on the predicted gene expression and see which trained model performed better. Quick 2 questions before we can discuss the evaluation of performance in more depth:


  1.
Is the predicted counts already in log-norm scale? Because we start from log-norm counts, so I wonder if the predicted values are already “normalised” or is there a normalisation of the predicted expression required prior to any downstream analysis?
  2.
Following the Xenium lung example, I have been training the model for inference on a sample-basis (each model = each sample). Can the trained model for inference be concatenated or trained from several samples at once?

Thank you!
Joao

Sent from Outlook for Mac
From: Kalin Nonchev ***@***.***>
Date: Thursday, 26 March 2026 at 06:25
To: ratschlab/DeepSpot ***@***.***>
Cc: Joao Luiz ***@***.***>, Mention ***@***.***>
Subject: Re: [ratschlab/DeepSpot] Employing DeepCell in COSMX data (Issue #17)

[https://avatars.githubusercontent.com/u/50597791?s=20&v=4]KalinNonchev left a comment (ratschlab/DeepSpot#17)<https://github.com/ratschlab/DeepSpot/issues/17#issuecomment-4132027846>

Hi @joaolsf<https://github.com/joaolsf>, thanks for the detailed update, great to see the results so far!

These are interesting questions, but to give meaningful suggestions, I would need to better understand the specifics of your setup and analysis goals. Given that this goes a bit beyond the scope of the repository, it might make sense to discuss this in more depth offline. Feel free to reach out via email if you're interested.

**Comment by KalinNonchev** [comment]
Hi @joaolsf , thank you for the questions. 

1. DeepSpot predicts log1p normalised gene expression. It is better not to normalize the data additionally because this introduces gene-gene dependencies. 
2. DeepSpot gets better with more data (more samples, better results). We show such a scaling experiment in Fig. 6B (https://www.medrxiv.org/content/10.1101/2025.02.09.25321567v3.full.pdf).

**Comment by joaolsf** [comment]
Hi Kalin,

Thanks for this clarification and sending the link of the paper. The version I have does not have figure 6 based on Xenium. Actually there is no mention of xenium in the version I had. I will see how I can train the model with a combination of samples.

Thanks,
Joao

Sent from Outlook for Mac
From: Kalin Nonchev ***@***.***>
Date: Thursday, 26 March 2026 at 10:24
To: ratschlab/DeepSpot ***@***.***>
Cc: Joao Luiz ***@***.***>, Mention ***@***.***>
Subject: Re: [ratschlab/DeepSpot] Employing DeepCell in COSMX data (Issue #17)

[https://avatars.githubusercontent.com/u/50597791?s=20&v=4]KalinNonchev left a comment (ratschlab/DeepSpot#17)<https://github.com/ratschlab/DeepSpot/issues/17#issuecomment-4133398020>

Hi @joaolsf<https://github.com/joaolsf> , thank you for the questions.

  1.  DeepSpot predicts log1p normalised gene expression. It is better not to normalize the data additionally because this introduces gene-gene dependencies.
  2.  DeepSpot gets better with more data (more samples, better results). We show such a scaling experiment in Fig. 6B (https://www.medrxiv.org/content/10.1101/2025.02.09.25321567v3.full.pdf).
